File: php/src/Model/TaskContext.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * The rich data stored against a task.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TaskContext
{
    public function __construct(
        public readonly string $taskId,
        public readonly ?string $title,
        public readonly ?string $description,
        /** @var array<string, mixed>|null */
        public readonly ?array $context,
        /** @var list<TaskReference> */
        public readonly array $references,
        public readonly ?int $createdAt,
        public readonly ?int $updatedAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            taskId: (string) ($data['taskId'] ?? ''),
            title: isset($data['title']) ? (string) $data['title'] : null,
            description: isset($data['description']) ? (string) $data['description'] : null,
            context: $data['context'] ?? null,
            references: Json::parseEach($data, 'references', [TaskReference::class, 'fromArray']),
            createdAt: isset($data['createdAt']) ? (int) $data['createdAt'] : null,
            updatedAt: isset($data['updatedAt']) ? (int) $data['updatedAt'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TaskContext.java
package io.fivexer.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * The rich data stored against a task: title, description, free-form context and references.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TaskContext {
    private String taskId;
    private String title;
    private String description;
    private Map<String, Object> context;
    private List<TaskReference> references;
    private Long createdAt;
    private Long updatedAt;

    public String getTaskId() { return taskId; }
    public String getTitle() { return title; }
    public String getDescription() { return description; }
    public Map<String, Object> getContext() { return context; }
    public List<TaskReference> getReferences() { return references; }
    public Long getCreatedAt() { return createdAt; }
    public Long getUpdatedAt() { return updatedAt; }
}
